<a href="https://colab.research.google.com/github/Mohdd-Afaan/data-science-master-2.0/blob/main/Statistics_Basic_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Q1. What are the three measures of central tendency?

Ans: The three measures of central tendency are:
Mean:
The mean is the average of all values in a dataset.
It is calculated by summing up all values and dividing by the total number of values.
Median:
The median is the middle value in a sorted dataset.
If the dataset has an odd number of observations, the median is the middle value. If the dataset has an even number of observations, the median is the average of the two middle values.
Mode:
The mode is the value that occur most frequently in a dataset.


Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

Ans: Differences:
Sensitivity to Outliers:
1)Mean is sensitive to outliers, as it considers all values in the dataset.
2)Median is resistant to outliers, as it is not influenced by extreme values.
3)Mode is not particularly sensitive to outliers, as it focuses on the most frequent values.
Calculation Method:
1)Mean involves summing all values and dividing by the total number of values.
2)Median involves sorting the data and finding the middle value.
3)Mode involves identifying the most frequent value(s).
Applicability:
1)Mean is suitable for symmetric distributions without outliers.
2)Median is suitable for skewed distributions and datasets with outliers.
3)Mode is suitable for identifying the most frequent values in any type of distribution.
Use in Measuring Central Tendency:
1)These measures provide a way to summarize the central or typical value of a dataset.
2)The choice of measure depends on the distribution of the data and the presence of outliers.
3)Mean, median, and mode are complementary and collectively offer insights into the characteristics of a dataset.

Q3. Measure the three measures of central tendency for the given height data:

In [10]:
import numpy as np
from scipy import stats

In [7]:
height = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [8]:
np.mean(height)

177.01875

In [9]:
np.median(height)

177.0

In [11]:
stats.mode(height)

ModeResult(mode=177.0, count=3)

Q4. Find the standard deviation for the given data:

In [12]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [16]:
np.std(data)

1.7885814036548633

Ans: Measures of Dispersion:
Range:
Definition: Range is the difference between the maximum and minimum values in a dataset.
Use: It provides a simple measure of the spread of data but is sensitive to outliers.
Variance:
Definition: Variance is the average of the squared differences between each data point and the mean of the dataset.
Use: It quantifies the overall variability of data. However, the squared units may make interpretation challenging.
Standard Deviation:
Definition: Standard deviation is the square root of the variance. It represents the average deviation from the mean.
Use: It is a more interpretable measure of the spread, allowing for a better understanding of the dispersion.


Example:
Consider the following dataset representing the daily temperatures in degrees Celsius for a city over a week:

In [32]:
temperature = [22,23,21,25,24,22,26]

In [64]:
def stats_range(x):
  return max(x) - min(x)

In [34]:
stats_range(temperature)

5

In [35]:
np.var(temperature)

2.7755102040816326

In [39]:
np.mean(temperature)

23.285714285714285

In [36]:
np.std(temperature)

1.6659862556700857

Q6. What is a Venn diagram?

Ans: A Venn diagram is a visual representation that illustrates the relationships and similarities between different sets or groups.  Venn diagrams use overlapping circles or other shapes to show the commonalities and differences among sets.

Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A intersection B
(ii) A ⋃ B

Ans:
(i)A intersection B={2,6}
(ii)A⋃B={0,2,3,4,5,6,7,8,10}

Q8. What do you understand about skewness in data?

Ans: Skewness is a statistical measure that describes the asymmetry of a probability distribution. In other words, it quantifies the extent and direction of skew (departure from horizontal symmetry) in a dataset. Skewness is an important concept in descriptive statistics and data analysis, as it provides insights into the shape of the distribution of data

Q9. If a data is right skewed then what will be the position of median with respect to mean?

Ans:
In a right-skewed distribution, also known as positively skewed distribution, the tail on the right side is longer or fatter than the left side. This implies that there are some extreme values (outliers) on the right side that are pulling the mean in that direction. In such a distribution:
Position of Mean: The mean is typically greater than the median.
Position of Median: The median is generally closer to the left tail and is less than the mean.

Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?

 Ans: Covariance:
 Covariance measures the degree to which two variables change together. If the covariance is positive, it indicates that as one variable increases, the other variable tends to increase as well. If the covariance is negative, it indicates that as one variable increases, the other variable tends to decrease.
 Correlation:
 Correlation is a standardized measure of the strength and direction of the linear relationship between two variables. It ranges from -1 to 1, where -1 indicates a perfect negative linear relationship, 1 indicates a perfect positive linear relationship, and 0 indicates no linear relationship.
 Differences:
 Scale: Covariance is not standardized, and its scale depends on the units of the variables being measured. Correlation, on the other hand, is standardized and always lies between -1 and 1.
 Interpretation: Correlation is more interpretable than covariance because it provides a standardized measure that is not influenced by the scale of the variables.
 Covariance: Covariance is used to understand how two variables change together. However, because its scale is dependent on the units of the variables, it can be difficult to compare covariances between different pairs of variables.
 Correlation: Correlation is widely used because of its standardized scale, making it easier to compare relationships between different pairs of variables.


Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.

Ans: sample mean = sum of terms/number of terms
dataset = [22,23,21,25,24,22,26]
sample mean = (22+23+21+25+24+22+26)/7
sample mean = 23.28571428571429


Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Ans: For a normal distribution, which is a symmetric bell-shaped distribution, the measures of central tendency—mean, median, and mode—are all the same. This is a unique characteristic of a perfectly normal distribution.

Q13. How is covariance different from correlation?

Ans: Covariance measures the direction of the relationship and is influenced by the scale of the variables.
Correlation provides a standardized measure, making it easier to compare and interpret relationships across different pairs of variable

Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Ans: Impact on Measures of Central Tendency:
Mean: Outliers can significantly influence the mean by pulling it towards extreme values. The mean is sensitive to outliers because it considers the magnitude of each data point.
Median: The median is less affected by outliers since it only depends on the order of values. A few extreme values don't impact the median as much.
Impact on Measures of Dispersion:
Range: Outliers can substantially affect the range, especially if they are far from the bulk of the data. The range is the difference between the maximum and minimum values.
Standard Deviation/Variance: Outliers increase the variability, leading to higher standard deviation and variance. They contribute to the squared differences from the mean in the calculation.

Consider the dataset:

In [60]:
without_outlier= [10,12,14,15,16,17,18,20]
with_outlier= [10,12,14,15,16,17,18,90]

central tendency

In [69]:
print("mean without outlier",np.mean(without_outlier))
print("mean with outlier",np.mean(with_outlier))
print("median without outlier",np.median(without_outlier))
print("median with outlier",np.median(with_outlier))

mean without outlier 15.25
mean with outlier 24.0
median without outlier 15.5
median with outlier 15.5


dispersion

In [72]:
print("range without outlier",stats_range(without_outlier))
print("range with outlier",stats_range(with_outlier))
print("variance without outlier",np.var(without_outlier))
print("variance with outlier",np.var(with_outlier))
print("standard deviation without outlier",np.std(without_outlier))
print("standard deviation with outlier",np.std(with_outlier))

range without outlier 10
range with outlier 80
variance without outlier 9.1875
variance with outlier 628.25
standard deviation without outlier 3.031088913245535
standard deviation with outlier 25.064915718988566
